Move features data and icons to Automattic data stores

alshakero opened the pull request:
#### Changes proposed in this Pull Request
* Extracted features data to `@automattic/data-store` package and feature icons to `@automattic/onboarding` package. With this change we are making them usable in both Gutenboarding and wp-admin (and anywhere else).
* Added the same redirect condition for Features step as it is currently implemented for Plans step (if no design is selected, then redirect to `/new`). 
  * This fixes a navigation bug in production. You can reproduce it by going to http://wordpress.com/new/features in an incognito window and try to continue to the next step. You'll be redirected to `/new` instead of landing on `/new/plans`.

#### Testing instructions
* Go to `/new` and advance to the `/new/features` step.
* Test feature picker functionality is the same as in production (originally introduced in https://github.com/Automattic/wp-calypso/pull/44431): 
  * by selecting a feature, a "Recommended" badge should  should be displayed  on the highlighted plan at `/new/plans` step.
  * If no feature is selected (step is skipped), `/new/plans` step shows "Popular" badge on the highlighted plan.
  * After creating a site, selected features should be resetted.
* Go to `/new/features?fresh`. You should be redirected to `/new`.

Fixes: https://github.com/Automattic/wp-calypso/issues/45213

razvanpapadopol commented on packages/data-stores/package.json:
```diff
@@ -44,6 +45,7 @@
 		"wpcom-proxy-request": "^6.0.0"
 	},
 	"peerDependencies": {
+		"react": "^16.13.1",
```
I was hoping we can avoid mixing data and components together as mentioned in the original issue:
> SVG icons can probably stay in Gutenboarding since they are used only in the <Features /> component.

A solution I see is to add a `FeatureIcon` component in Gutenboarding for now (we can move it anytime to `automattic/onboarding` package if we need it elsewhere) that takes a feature slug as prop and renders the correct icon.

simison replied:
I paused for a moment with this as well. :-)

Since these are a static list of labels/slugs/icons, I think I'd expect them to be in somewhere else than data store and simply imported to plans grid. No super strong feeling about it tho, and I like Razvan's suggestion.

Data store would then be responsible of moving around feature slugs only e.g. as storing selected features.

So visual representation (icons, and I'd argue also translatable label, and descriptions) would be separate from the data (i.e. slugs).

alshakero replied:
Yes, I saw that in the issue but I figured the plans page will be using these icons soon (p1598338944149400-slack-CPFFQTG69). But tbh, it didn't occur to me that this package is **data**-stores. I'll put them in the `automattic/onboarding` package.

razvanpapadopol replied:
Now I see the icons next to the features in the new plans grid design 👍 
Having them in `automattic/onboarding` makes sense then!

simison commented:
The text alignment fix would make a good another PR but no strong feeling about having it combined here either.

alshakero commented:
I think this is ready for another round. I cherry-picked the alignment fix to [another PR](https://github.com/Automattic/wp-calypso/pull/45249) and addressed all feedback.

razvanpapadopol commented on client/landing/gutenboarding/stores/onboard/reducer.ts:
```diff
@@ -239,7 +223,6 @@ const reducer = combineReducers( {
 	siteVertical,
 	showSignupDialog,
 	plan,
-	selectedFeatures,
```
The user selection of features should probably belong to Onboard store (same as selected vertical/fonts/design/domain, etc). In order to pass these to Site Setup flow we'll use site options (see related issue https://github.com/Automattic/wp-calypso/issues/45148) and they will become available [on the SITE_STORE](https://github.com/Automattic/wp-calypso/blob/master/apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/hooks/use-site.ts#L18).

razvanpapadopol reviewed (commented):
The changes are looking good to me with just [one small note](https://github.com/Automattic/wp-calypso/pull/45221#discussion_r479358281). 
I'll leave it now for @yansern to give the final ✅

wp-desktop reviewed (changes requested):
WordPress Desktop CI Failure for job "wp-desktop-mac".

@razvanpapadopol please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/891185). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

razvanpapadopol commented on packages/onboarding/src/feature-icon/index.tsx:
```diff
@@ -0,0 +1,100 @@
+
+const FeatureIcon: React.FunctionComponent< Props > = function FeatureIcon( {
+	featureId,
+	...props
```
Implicitly passed props often lead to unexpected results. Let's just explicitly handle these when we need to pass something. Right now I see only `featureId` is being used.

razvanpapadopol replied:
Fixed in https://github.com/Automattic/wp-calypso/pull/45221/commits/512c62e27be45479aeea94673c669c2fb4776d3e
cc: @alshakero

ciampo reviewed (approved):
LGTM  🙌
